contrib/babelfishpg_tsql/src/pl_handler.c

```diff
@@ -362,40 +379,61 @@ assign_identity_insert(const char *newval, void *extra)
 		option_flag = (char *) linitial(elemlist);
 		rel_name = (char *) lsecond(elemlist);
 
+		/* Use catalog name if provided */
+		if (list_length(elemlist) == 4)
+		{
+			catalog_name = (char *) lfourth(elemlist);
+
+			if(!DbidIsValid(get_db_id(catalog_name)))
+			{
+				/* Get schema name for error message */
+				logical_schema_name = (char *) lthird(elemlist);
+				throw_error_for_identity_insert(catalog_name, logical_schema_name, rel_name);
+			}
+
+			cur_db_name = catalog_name;
+			curr_user_if_cross_db = get_user_for_database(cur_db_name);
+
+			/*
+			 * User in cross-db case can be NULL if login has no user
+			 * that it can use to connect to that database. We should
+			 * throw permission denied error in that case
+			 */
+			if (!curr_user_if_cross_db)
+				throw_error_for_identity_insert(catalog_name, logical_schema_name, rel_name);
+		}
+
 		/* Check the user provided schema value */
 		if (list_length(elemlist) >= 3)
 		{
-			schema_name = (char *) lthird(elemlist);
+			logical_schema_name = (char *) lthird(elemlist);
 
 			if (cur_db_name)
 				schema_name = get_physical_schema_name(cur_db_name,
-													   schema_name);
+													   logical_schema_name);
+
+			/* If no schema name is provided, we should use default schema */
+			if (!schema_name)
```

Comment:
By this logic, We will be using default schema when user supplies wrong schema.

Reply:
Do we have any testcase covering it?

Reply:
`get_physical_schema_name()` doesn't perform the catalog look up. The actual error will get thrown at https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3402/files#diff-ce1101bab57d0e7f8b36f0df103ed3ae49f4b488e473ac87bf2320c32f2697e1R430. For test case see https://github.com/babelfish-for-postgresql/babelfish_extensions/pull/3402/files#diff-c16577e6f7e7da2222539138bf10997966a06ded7b507a823fe31328fbfe27d4R146.